File: Maths/falling_asleep.py
# AtCoder Dwango Programming Contest (Problem A)

class Solution(object):
    def music(self,s_arr:list,k_arr:list, X:str, N:int)->int:
        self.count = 0
        for i in range(N):
            if(s_arr[i]==X):
                for j in range(i+1,N):
                    self.count+=k_arr[j]
        return self.count


if __name__=='__main__': # test program
    N = 3
    s_arr = ['dwango','sixth','roll']
    k_arr = [2,5,25]
    X = 'dwango'
    s = Solution()
    print(s.music(s_arr,k_arr,X,N)) # 30

File: Maths/falling_asleep.cxx
// AtCoder Dwango Programming Contest 6th (Problem A)

#include <bits/stdc++.h>
using namespace std;

class Solution{
    public:
    static int music(vector<pair<string,int>> pairvec, string &X){
        int count = 0;
        for(auto i = 0; i < pairvec.size(); i++){
            if(pairvec[i].first == X){
                for(auto j = i+1; j < pairvec[j].second; j++)
                    count+=pairvec[j].second;
            }
        }
        return count;
    }
};

int main(){
    ios_base::sync_with_stdio(0);
    cout.tie(0); cin.tie(0);

    vector<pair<string,int>> pairvec;
    string s, X; int k;
    int count = 0;
    int N; cin >> N;

    for (auto i = 0; i < N; i++){
        cin >> s >> k;
        pairvec.emplace_back(s,k);
    }

    cin >> X;
    cout << Solution::music(pairvec,X);
    return 0;
}
